image/preseed/preseed_linux.go

```diff
@@ -217,13 +217,17 @@ func prepareCore20Mountpoints(prepareImageDir, tmpPreseedChrootDir, snapdSnapBlo
 
 	var mounted []string
 
+	umountCmd := func(mnt string) {
+		cmd := exec.Command("umount", mnt)
+		if out, err := cmd.CombinedOutput(); err != nil {
+			fmt.Fprintf(Stdout, "cannot unmount: %v\n'umount %s' failed with: %s", err, mnt, out)
+		}
+	}
+
 	cleanupMounts = func() {
-		for i := len(mounted) - 1; i >= 0; i-- {
+		for i := len(mounted) - 1; i > 0; i-- {
 			mnt := mounted[i]
-			cmd := exec.Command("umount", mnt)
-			if out, err := cmd.CombinedOutput(); err != nil {
-				fmt.Fprintf(Stdout, "cannot unmount: %v\n'umount %s' failed with: %s", err, mnt, out)
-			}
+			umountCmd(mnt)
```

Comment:
(nitpick^2) I would call this either just `unmount()` or `doUnmount()` or something. This looks a bit like it would just return an exec.Cmd